src/sentry/api/endpoints/organization_details.py

```diff
@@ -116,6 +123,48 @@ def update_organization_scenario(runner):
         )
 
 
+class TrustedRelaySerializer(serializers.Serializer):
+    def to_representation(self, instance):
+        return convert_dict_key_case(instance, snake_to_camel_case)
+
+    def to_internal_value(self, data):
+
+        key_name = "{unknown}"
+        try:
+            key_name = data.get(u"name")
+            public_key = data.get(u"publicKey")
+
+            PublicKey.parse(public_key)
+
+            if key_name is None:
+                raise serializers.ValidationError("missing relay key name")
+
+            key_name = key_name.strip()
+
+            if len(key_name) == 0:
+                raise serializers.ValidationError(
+                    "invalid relay key name, must be non empty string"
+                )
+
+            ret_val = {
+                u"public_key": public_key,
+                u"name": key_name,
+                u"description": data.get(u"description"),
+            }
+
+            return ret_val
+        except serializers.ValidationError:
+            raise
+        except RelayError as e:
+            raise serializers.ValidationError(
+                "Invalid relay key for trusted relay:{}, e:{}".format(key_name, e)
+            )
```

Comment:
Can you reorganize this try-catch? 

- There's no need for catching and re-raising `ValidationError`.
- `RelayError` only happens in one specific spot around `PublicKey.parse`. It would be nice to wrap just that.
- We usually don't have _catch-all_ excepts. If there are other errors we do not anticipate, it's fine to let it go through and cause a 500. That will allow us to see the exact location where this happens in Sentry later.

Reply:
I agree that we should let 500 pass through. 
The code here is only parsing the data though (no db access or other things like that that would generate something that should be a `HTTP 500`).

The only case where the` Exception` branch is executed is when the `value` parameter is not a dictionary (and we do a  `value.get(u"name")`) .

I've partitioned the code in multiple try blocks and changed the except clause to the more specific `AttributeError`.